portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,19 +43,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(convertToJson(messages));
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /** Get the message from the form and add it to the array */
+    String message = request.getParameter("text-input");
+    messages.add(message);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
   /**
    * Converts an array into a JSON string using manual String concatentation.
    */
   private String convertToJson(ArrayList messages) {
     String json = "{";
-    json += "\"string1\": ";
-    json += "\"" + messages.get(0) + "\"";
-    json += ", ";
-    json += "\"string2\": ";
-    json += "\"" + messages.get(1) + "\"";
-    json += ", ";
-    json += "\"string3\": ";
-    json += "\"" + messages.get(2) + "\"";
+    int i;
+    for (i = 0; i < messages.size() - 1; i++) {
+      json += "\"string" + i + "\": ";
+      json += "\"" + messages.get(i) + "\"";
+      json += ", ";
+    }
+    json += "\"string" + i + "\": ";
+    json += "\"" + messages.get(i) + "\"";
```

Comment:
This looks like to me the last message will get added twice, is that intentional?

Reply:
It does not get added twice (the condition is set to messages.size() - 1), but I do add the last message separately. This is because the last object doesn't go with a comma at the end.